splice_machine/src/main/java/com/splicemachine/derby/lifecycle/EngineLifecycleService.java

```diff
@@ -92,13 +95,44 @@ public void start() throws Exception{
         // External connections to Derby are created later when the Derby network server is started.
         EmbedConnectionMaker maker = new EmbedConnectionMaker();
         toUpgrade.set(isMaster);
-        if(startup.connectAsFirstTime()){
-            isCreate.set(Boolean.TRUE);
-            internalConnection=maker.createFirstNew(configuration,dbProperties);
-        }else{
-            isCreate.set(Boolean.FALSE);
-            internalConnection=maker.createNew(dbProperties);
-        }
+        boolean onHold = false;
```

Comment:
I don't think this is the right place to be retrying, I'm still trying to think what that place should be.

Reply:
I've tried to catch the error in `SpliceMasterObserver` but `EngineLifecycleService` is started by `DatabaseLifecycleManager`, which cannot catch the error. Other possible places are:

```
com.splicemachine.derby.impl.store.access.SpliceAccessManager.boot(SpliceAccessManager.java:636)
        at com.splicemachine.db.impl.services.monitor.BaseMonitor.boot(BaseMonitor.java:1963)
        at com.splicemachine.db.impl.services.monitor.TopService.bootModule(TopService.java:337)
        at com.splicemachine.db.impl.services.monitor.BaseMonitor.startModule(BaseMonitor.java:546)
        at com.splicemachine.db.impl.services.monitor.FileMonitor.startModule(FileMonitor.java:57)
        at com.splicemachine.db.iapi.services.monitor.Monitor.bootServiceModule(Monitor.java:435)
        at com.splicemachine.derby.impl.db.SpliceDatabase.bootStore(SpliceDatabase.java:443)
        at com.splicemachine.db.impl.db.BasicDatabase.boot(BasicDatabase.java:170)
        at com.splicemachine.derby.impl.db.SpliceDatabase.boot(SpliceDatabase.java:120)
        at com.splicemachine.db.impl.services.monitor.BaseMonitor.boot(BaseMonitor.java:1963)
        at com.splicemachine.db.impl.services.monitor.TopService.bootModule(TopService.java:337)
        at com.splicemachine.db.impl.services.monitor.BaseMonitor.bootService(BaseMonitor.java:1798)
        at com.splicemachine.db.impl.services.monitor.BaseMonitor.startProviderService(BaseMonitor.java:1665)
        at com.splicemachine.db.impl.services.monitor.BaseMonitor.findProviderAndStartService(BaseMonitor.java:1543)
        at com.splicemachine.db.impl.services.monitor.BaseMonitor.startPersistentService(BaseMonitor.java:967)
        at com.splicemachine.db.iapi.services.monitor.Monitor.startPersistentService(Monitor.java:557)
        at com.splicemachine.db.impl.jdbc.EmbedConnection.bootDatabase(EmbedConnection.java:2278)
```

If we push the catch into deeper methods, like these ones:

```
com.splicemachine.derby.utils.ConglomerateUtils.readConglomerate(ConglomerateUtils.java:106)
        at com.splicemachine.derby.impl.store.access.hbase.HBaseConglomerateFactory.readConglomerate(HBaseConglomerateFactory.java:135)
        at com.splicemachine.derby.impl.store.access.SpliceAccessManager.conglomCacheFind(SpliceAccessManager.java:164)
        at com.splicemachine.derby.impl.store.access.SpliceTransactionManager.findExistingConglomerate(SpliceTransactionManager.java:193)
        at com.splicemachine.derby.impl.store.access.SpliceTransactionManager.openScan(SpliceTransactionManager.java:962)
        at com.splicemachine.derby.impl.store.access.PropertyConglomerate.openScan(PropertyConglomerate.java:178)
        at com.splicemachine.derby.impl.store.access.PropertyConglomerate.readStoreProperties(PropertyConglomerate.java:610)
        at com.splicemachine.derby.impl.store.access.PropertyConglomerate.readProperties(PropertyConglomerate.java:588)
        at com.splicemachine.derby.impl.store.access.PropertyConglomerate.getCachedProperties(PropertyConglomerate.java:650)
        at com.splicemachine.derby.impl.store.access.PropertyConglomerate.<init>
```

Then code other than boot process may call these methods, which we want to throw an error if the database is booted but cannot get conglomerate.

Reply:
I would start by retrying the test with a larger timeout, 10ms is too low, try 10000 or 30000. If it this happens again we can think where's the best place to retry. Currently it's not retrying even once due to the very low timeout